Tools/gdb/libpython.py

```diff
@@ -1874,16 +1874,47 @@ def invoke(self, args, from_tty):
             print('Unable to read information on python frame')
             return
 
-        pyop_var, scope = pyop_frame.get_var_by_name(name)
-
-        if pyop_var:
+        obj = None
+        fields = name.split('.')
+        for field in fields:
+            if obj is None:
+                obj, scope = pyop_frame.get_var_by_name(field)
+            else:
+                try:
+                    obj = self.get_object_field(obj, field)
+                except LookupError:
+                    print("Can't find field %s of %s" % (field, obj))
+                    return
+        if obj:
             print('%s %r = %s'
                    % (scope,
                       name,
-                      pyop_var.get_truncated_repr(MAX_OUTPUT_LEN)))
+                      obj.get_truncated_repr(MAX_OUTPUT_LEN)))
         else:
             print('%r not found' % name)
 
+    'Based on type of object, return field of the object'
+    def get_object_field(self, obj, field):
+        if isinstance(obj, PyListObjectPtr) or isinstance(obj, PyTupleObjectPtr):
+            # field must be an integer index
```

Comment:
It seems a bit odd to be using dotted notation to access by index.  Is it possible to support square-bracket syntax here, or does it conflict with gdb's usage of square-bracket syntax for C level lookup?